reader_test.go

```diff
@@ -296,6 +296,167 @@ func TestDecodeMediaPlaylistAutoDetectExtend(t *testing.T) {
 	}
 }
 
+// Tests for Time Parsing of EXT-X-PROGRAM-DATE-TIME
+// We testing ISO/IEC 8601:2004 where we can get time in UTC, UTC with Nanoseconds
+// timeZone in formats '±00:00', '±0000', '±00'
+// m3u8.FullTimeParse()
+func TestTimeLayoutsDecode(t *testing.T) {
```

Comment:
This test is really just testing the `FullTimeParse` function, but it's testing it via `decodeLineOfMediaPlaylist`, I'd just call this function directly.

Also, there's a lot of repetition in the tests, can you rewrite using https://github.com/golang/go/wiki/TableDrivenTests (or sub tests) similar to https://github.com/grafov/m3u8/blob/f55aa7a0ae5bfe63e82ee5adefa3aa4e8692b3dd/writer_test.go#L223-L253 ?